### ISSUE TITLE
Interactive plotting

### ISSUE THREAD

**Issue by f0uriest** [ddudt/DESC#40]
Right now we only plot stuff at the end of the program. I might be nice to have an option to plot the latest results after each outer iteration, or to plot the initial guess before starting, etc. We could make the plot flag have a value, like ``plot=2`` with different levels, like with ``verbose`` 

By default matplotlib blocks execution when plotting (ie, waiting for the user to close the plot before continuing the program), which we might want or might not. There are a few different options for handling that, but I think we probably could make it so it doesn't block and it will just display plots when ready and keep on calculating

**Comment by f0uriest** [comment]
Given that most runs will be done on clusters without GUI support I don't think we need this.
